generate/hcl.go

```diff
@@ -277,3 +308,134 @@ func checkHCLProviders(mod *configs.Module, opt Options) (Options, error) {
 
 	return opt, nil
 }
+
+// moduleInstall will recursively walk through the module calls required by the Terraform config, it will store the downloaded module
+// in $XDG_CACHE directory and stop once all the required modules have been downloaded.
+func moduleInstall(calls []*configs.ModuleCall, mRes *map[string]*configs.Resource, pwd string, installedModules map[string]struct{}) error {
+	// stop condition, if there is no module to
+	// fetch we stop
+	if len(calls) == 0 {
+		return nil
+	}
+
+	call := calls[0]
+	name := call.Name
+
+	// we check if the module is already installed
+	// or not
+	if _, ok := installedModules[name]; ok {
+		return nil
+	}
+
+	var src string = call.SourceAddr
+
+	// we check if the module is a Terraform registry module
+	// in order to get its source address from Terraform registry
+	if regMod, err := regsrc.ParseModuleSource(src); err == nil {
+		client := registry.NewClient(nil, nil)
+		// we get the list of available module versions
+		resp, err := client.ModuleVersions(regMod)
+		if err != nil {
+			return fmt.Errorf("unable to get module versions: %w", err)
+		}
+
+		if len(resp.Modules) < 1 {
+			return fmt.Errorf("unable to find suitable versions")
+		}
+		meta := resp.Modules[0]
+
+		var (
+			latest *version.Version
+			// match holds the version matching the
+			// source constraints set in the module call
+			match *version.Version
+		)
+		for _, vers := range meta.Versions {
+			v, err := version.NewVersion(vers.Version)
+			if err != nil {
+				return fmt.Errorf("unable to create version from string: %w", err)
+			}
+
+			if latest == nil || v.GreaterThan(latest) {
+				latest = v
+			}
+
+			if call.Version.Required.Check(v) {
+				if match == nil || v.GreaterThan(match) {
+					match = v
+				}
+			}
+		}
+		// we finally get the module location, it will return
+		// a string `go-getter` compliant
+		src, err = client.ModuleLocation(regMod, match.String())
+		if err != nil {
+			return fmt.Errorf("unable to fetch module location: %w", err)
+		}
+	}
+
+	// since go-getter does not support yet in-memory fs,
+	// we need to initialize the parser using actual fs
+	// https://github.com/hashicorp/go-getter/issues/83
+	pars := configs.NewParser(nil)
+
+	// we check if the module is a local one by checking
+	// its prefix "./", "../", etc.
+	var isLocal bool
+	for _, prefix := range localSourcePrefixes {
+		if strings.HasPrefix(src, prefix) {
+			isLocal = true
+		}
+	}
```

Comment:
Isnt `source = "my-module"` local too? Or is it converted

Reply:
According to the [doc](https://www.terraform.io/docs/language/modules/sources.html#local-paths):
> A local path must begin with either ./ or ../ to indicate that a local path is intended, to distinguish from a module registry address.